tests/test_core.py

```diff
@@ -3,82 +3,64 @@
 import numpy as np
 from turbopy.core import Grid
 
-def test_grid():
-    """Test initialization of the Grid class"""
-    n_grid = 8
-    grid_conf = {"N": n_grid,
+#Grid class test methods
+@pytest.fixture(name='conf')
+def grid_conf():
+    """Pytest fixture for grid configuration dictionary"""
+    grid_dict = {"N": 8,
                  "r_min": 0,
                  "r_max": 0.1}
-    grid = Grid(grid_conf)
-    assert grid.grid_data == grid_conf
+    return grid_dict
+
+def test_grid(conf):
+    """Test initialization of the Grid class"""
+    grid = Grid(conf)
+    assert grid.grid_data == conf
     assert grid.r_min == 0.0
     assert grid.r_max == 0.1
```

Comment:
If you change `conf` to `simple_grid`, then you can write:
```suggestion
def test_grid(simple_grid):
    """Test initialization of the Grid class"""
    assert simple_grid.r_min == 0.0
    assert simple_grid.r_max == 0.1
```
Note that you don't have to construct the `Grid` object, and you also don't have to check the `grid_data` dictionary (since that was already checked in the fixture).

Reply:
Why don't I have to construct the Grid object? I'm testing the init method.

Reply:
You are correct that you should test the `__init__` method, and for that you should be constructing the Grid object. What I was should have made more clear is that for all the other tests, you can simplify them by constructing the grid object in the fixture. Does that make sense?

Reply:
Ah yes that makes sense. Thanks